seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/common/serialization/InternalRowConverter.java

```diff
@@ -111,23 +128,35 @@ private static InternalRow convert(SeaTunnelRow seaTunnelRow, SeaTunnelRowType r
 
     private static Object convertMap(Map<?, ?> mapData, MapType<?, ?> mapType, BiFunction<Object, SeaTunnelDataType<?>, Object> convertFunction) {
         if (mapData == null || mapData.size() == 0) {
-            return mapData;
+            return ArrayBasedMapData.apply(new Object[]{}, new Object[]{});
         }
-        switch (mapType.getValueType().getSqlType()) {
-            case MAP:
-            case ROW:
-            case DATE:
-            case TIME:
-            case TIMESTAMP:
-                Map<Object, Object> newMap = new HashMap<>(mapData.size());
-                mapData.forEach((key, value) -> {
-                    SeaTunnelDataType<?> valueType = mapType.getValueType();
-                    newMap.put(key, convertFunction.apply(value, valueType));
-                });
-                return newMap;
-            default:
-                return mapData;
+        Map<Object, Object> newMap = new HashMap<>(mapData.size());
+        mapData.forEach((key, value) -> {
+            SeaTunnelDataType<?> keyType = mapType.getKeyType();
+            SeaTunnelDataType<?> valueType = mapType.getValueType();
+            newMap.put(convertFunction.apply(key, keyType), convertFunction.apply(value, valueType));
+        });
+        Object[] keys = newMap.keySet().toArray();
+        Object[] values = newMap.values().toArray();
+        return ArrayBasedMapData.apply(keys, values);
+    }
+
+    private static Object reconvertMap(MapData mapData, MapType<?, ?> mapType, BiFunction<Object, SeaTunnelDataType<?>, Object> convertFunction) {
+        if (mapData == null || mapData.numElements() == 0) {
+            return Collections.emptyMap();
+        }
+        Map<Object, Object> newMap = new HashMap<>(mapData.numElements());
+        int num = mapData.numElements();
+        SeaTunnelDataType<?> keyType = mapType.getKeyType();
+        SeaTunnelDataType<?> valueType = mapType.getValueType();
+        Object[] keys = mapData.keyArray().toObjectArray(seaTunnelType2SparkType(keyType));
+        Object[] values = mapData.valueArray().toObjectArray(seaTunnelType2SparkType(valueType));
+        for (int i = 0; i < num; i++) {
+            keys[i] = convertFunction.apply(keys[i], keyType);
+            values[i] = convertFunction.apply(values[i], valueType);
+            newMap.put(keys[i], values[i]);
         }
```

Comment:
Can't it be like this here?

```java
        Map<Object, Object> newMap = new HashMap<>(mapData.size());
        mapData.forEach((key, value) -> {
            SeaTunnelDataType<?> keyType = mapType.getKeyType();
            SeaTunnelDataType<?> valueType = mapType.getValueType();
            newMap.put(convertFunction.apply(key, keyType), convertFunction.apply(value, valueType));
        });
```

Reply:
If it can't, remove the `convertFunction` parameter in `#convertMap` and `#reconvertMap`, as it was meant to support both convert and reconvert, which is useless now.